src/main/java/com/googlecode/yatspec/junit/SpecListener.java

```diff
@@ -71,12 +72,21 @@ public void afterAll(ExtensionContext extensionContext) {
     }
 
     protected Optional<TestState> getOptionalTestState(Object testInstance) {
-        return Arrays.stream(testInstance.getClass().getDeclaredFields())
-                .filter(field -> field.getType() == TestState.class)
-                .map(field -> getField(testInstance, field))
-                .filter(Objects::nonNull)
-                .map(TestState.class::cast)
-                .findFirst();
+        Class<?> clazz = testInstance.getClass();
+        while (clazz != null) {
+            Optional<TestState> optionalField = Optional.ofNullable(clazz)
+                    .map(Class::getDeclaredFields)
+                    .stream().flatMap(Stream::of)
+                    .filter(field -> field.getType() == TestState.class)
+                    .map(field -> getField(testInstance, field))
+                    .filter(Objects::nonNull)
+                    .map(TestState.class::cast)
+                    .findFirst();
+
+            if (optionalField.isEmpty()) clazz = clazz.getSuperclass();
+            else return optionalField;
+        }
+        return Optional.empty();
```

Comment:
Maybe extracting the logic into a method to reuse would work? I'm not sure if i've broken functionality as I can't get the tests to run but it looks like it should work.

```suggestion
        Class<?> clazz = testInstance.getClass();
        return Optional.ofNullable(clazz)
                .flatMap(this::getTestStateForClass)
                .flatMap(testState1 -> getTestStateForClass(clazz.getSuperclass()));
    }

    private Optional<TestState> getTestStateForClass(Class<?> clazz) {
        return Optional.of(clazz)
                .map(Class::getDeclaredFields)
                .stream().flatMap(Stream::of)
                .filter(field -> field.getType() == TestState.class)
                .map(field -> getField(testInstance, field))
                .filter(Objects::nonNull)
                .map(TestState.class::cast)
                .findFirst();
```

Reply:
Not sure if i'm following; I think this change would fail my test as it is only going 2 deep into the hierarchy where my test is setup to go further than the initial test classes super class.

```
class NestedSuperClass extends AnotherClass { }
class AnotherClass extends ClassWithTestState { }
class WithTestState { private TestState ... }  
```

Reply:
Nice one guys! (sorry I have not had a chance to take a closer look yet - I intend to spend some time this week on Yatspec so will be able to take a closer look soon!)